### ISSUE TITLE
Change "readable" to "readonly" and "writeable" to "writable" in new globals configuration

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#11359]
**The version of ESLint you are using.**

master

**The problem you want to solve.**

#11338 merged a change that allowed globals to be specified as "writeable" or "readable". I believe "readable" doesn't accurately convey the setting. All variables present in the environment are readable, including writable variables

Also, "writeable" is misspelled. It should be spelled "writable".

**Your take on the correct solution to problem.**

1. Change "readable" to "readonly".
1. Change "writeable" to "writable".

**Are you willing to submit a pull request to implement this change?**

Yes, if no one else wants to.

**Comment by platinumazure** [comment]
I think the intention for "writeable" was to match the scope analysis property name, even though it is misspelled. (See [this discussion](https://github.com/eslint/eslint/pull/11338#discussion_r252497158) for more info.)

I'm okay with switching to "readonly".

**Comment by not-an-aardvark** [comment]
Yes, the spelling of "writeable" is intentional. "Writeable" is a [valid alternate spelling](https://en.wiktionary.org/wiki/writeable) of the word "writable", although it is less common.

"Readable" was also discussed in https://github.com/eslint/eslint/issues/9940#issuecomment-400103341.

**Comment by platinumazure** [comment]
@not-an-aardvark I don't think we really had consensus around "readable"-- it was offered as an option for consistency with writeable, but it wasn't ever really discussed one way or the other. Unfortunately, I didn't think to raise the question when reviewing the PR.

I do like "readonly" (with no punctuation) better than "readable".

**Comment by edg2s** [comment]
+1 to "readonly", as "readable" by itself doesn't mean "also, not writeable".

Strictly speaking "writ(e)able" by itself doesn't mean "also readable", and so you could use "readwrite", but I think in context it is probably clear that writ(e)able variables are always readable.

**Comment by not-an-aardvark** [comment]
At this point it doesn't seem ideal to bikeshed over names, given that the feature has already shipped and our options are limited by compatibility requirements.

Personally, I think "readable" and "writeable" are sufficiently clear to imply the correct meaning. I have no objection to adding aliases like "readonly", "readwrite", and "writable", although I'm also not convinced that they're really needed, and they might just lead to more confusion if a user sees "readwrite" and "writable" in a config and assumes they mean something different.

The potential for typos will be smaller if we disallow unexpected strings in this position in the next major release.

**Comment by not-an-aardvark** [comment]
Relabeling as "enhancement" because the current behavior is working as intended.

**Comment by nzakas** [comment]
My apologies, I actually didn't realize this feature had shipped when I opened this issue.

I understand the intention behind "writeable", but I also think that's significant cognitive overhead for end users (especially when I see red squigglies everywhere I type "writeable"), and I did think we had decided on "readonly" previously. I'm sorry I haven't been paying close enough attention to PRs as they come through, otherwise I wouldn't have mentioned all of this earlier. I'll do better in the future.

What I'd like to suggest is that we:

1. Add "readonly" as an alias for "readable"
1. Add "writable" as an alias for "writeable"
1. Change the docs to use "readonly" and "writable" while mentioning that "readable" and "writeable" are deprecated.
